src/DynamoCoreWpf/Views/Core/NoteView.xaml.cs

```diff
@@ -152,5 +168,268 @@ private void PrepareZIndex()
+        /// <param name="caretIndex"> caret index where user typed TAB </param>
+        /// <returns></returns>
+        private string BulletTabHandler(string text, int caretIndex)
+        {
+            // Get the line where TAB was typed 
+            var lineNumber = GetLineNumberAtCaretsIndex(text, caretIndex);
+            var line =  GetLineTextAtCaretsIndex(text,caretIndex);
+            var caretAtLine = GetCaretIndexRelativeToLine(text, caretIndex);
+
+            // If TAB was pressed just after a bullet insert it before the bullet
+            // so that you can easily indent
+            var distanceFromBulletToCaret = SPACING_AFTER_BULLET + 1;
+            var caretIsRightAfterBullet = IsCaretRightAfterBullet(line, caretAtLine, distanceFromBulletToCaret);
+            if (caretIsRightAfterBullet)
+            {
+                line = line.Insert(caretAtLine - distanceFromBulletToCaret, "\t");
+            }
+                
+            else
+            {
+                line = line.Insert(caretAtLine, "\t");
+                caretAtLine++;
+            }
+
+            // Also update the bullet character if CaretIndex is right after bullet or before bullet
+            if (IsCaretBeforeABullet(line, caretAtLine) || caretIsRightAfterBullet)
+            {
+                line = UpdateBulletAccordingToIndentation(line);
+            }
+                
+            return ReplaceLineOfText(text, lineNumber, line);
+        }
+
+        /// <summary>
+        /// Handles text when user types Key.Enter (ENTER)
+        /// When ENTER is pressed, if previous line had a bullet copy it to next line
+        /// </summary>
+        /// <param name="text">  text before typing ENTER </param>
+        /// <param name="caretIndex"> caret index where user typed ENTER </param>
+        /// <returns></returns>
+        private string BulletEnterHandler(string text, int caretIndex)
+        {
+            var lineNumber = GetLineNumberAtCaretsIndex(text, caretIndex);
+            var line = GetLineTextAtCaretsIndex(text, caretIndex);
+            var caretAtLine = GetCaretIndexRelativeToLine(text, caretIndex);
+
+            // Check if line has any bullets, and copy it to next line
+            var bulletsInLine = BULLETS_CHARS.Where(b => line.Contains(b));
+            if (bulletsInLine.Count() == 0)
+            {
+                return text = text.Insert(caretIndex, "\n");
+            }
+            
+            text = text.Insert(caretIndex, "\n");
+            caretIndex++;
+
+            var tabsBeforeBullet = line.Split(BULLETS_CHARS)[0].Count(c => c=='\t');
+            for (int i = 0; i < tabsBeforeBullet; i++)
+            {
+                text = text.Insert(caretIndex, "\t");
+                caretIndex++;
+            }
+            return text.Insert(caretIndex, bulletsInLine.First()+ new String(' ', SPACING_AFTER_BULLET));
+        }
+
+        private bool StringIsEmptyOrTab(string text)
```

Comment:
Hmm by reading the code, seems we are checking if the string input is a combination of spaces or tabs? If that is the case, maybe 
- name this `IsStringSpaceswithTabs`
- use text.All(c=>c==' ' or c=='\t')
Or maybe I read it wrong? Please elaborate

Reply:
ok, I agree, `IsStringSpaceswithTabs` is a better name and `String.IsEmpty(text) || text.All(c=>c==' ' or c=='\t')` avoids using a double negative